xarray_schema/dataset.py

```diff
@@ -0,0 +1,75 @@
+from typing import Any, Callable, Dict, Hashable, Iterable, Union
+
+import xarray as xr
+
+from .base import BaseSchema, SchemaError
+from .dataarray import DataArraySchema
+
+
+class DatasetSchema(BaseSchema):
+    '''A light-weight xarray.Dataset validator
+
+    Parameters
+    ----------
+    data_vars : mapping of variable names and DataArraySchemas, optional
+        Per-variable DataArraySchema's, by default None
+    checks : Iterable[Callable], optional
+        Dataset wide checks, by default None
+    '''
+
+    def __init__(
+        self,
+        data_vars: Dict[Hashable, Union[DataArraySchema, None]] = None,
```

Comment:
Could any of these be abstracted/shared in .typing?

Reply:
I think so but I wasn't sure which ones I'd use more than once. I'll take another pass through the type definitions.